tests/src/test/java/com/maxio/advancedbilling/controllers/subscriptiongroups/SubscriptionGroupsControllerAddRemoveTest.java

```diff
@@ -0,0 +1,212 @@
+        subscriptionIds.remove(signupResponse.getPrimarySubscriptionId());
+        return subscriptionIds.get(0);
+    }
+
+    @Nested
+    public class Add {
+
+        @Test
+        void shouldAddSecondarySubscriptionToGroup() throws IOException, ApiException {
+            // given
+            groupSignup = new TestSetup().signupWithSubscriptionGroup(product, meteredComponent);
+            Customer groupCustomer = CLIENT.getCustomersController().readCustomer(groupSignup.getCustomerId()).getCustomer();
+            assertThat(groupSignup.getSubscriptionIds())
+                    .hasSize(2);
+            Subscription newSubscription = new TestSetup().createSubscription(groupCustomer, product);
+            CreditCardPaymentProfile signupPP = CLIENT.getPaymentProfilesController()
+                    .readPaymentProfile(groupSignup.getPaymentProfileId())
+                    .getPaymentProfile()
+                    .match((new PaymentProfileResponsePaymentProfileGetter<>()));
+
+            // when
+            SubscriptionGroup response = SUBSCRIPTION_GROUPS_CONTROLLER.addSubscriptionToGroup(newSubscription.getId(), new AddSubscriptionToAGroup(
+                    new GroupSettings.Builder()
+                            .target(new GroupTarget(GroupTargetType.SUBSCRIPTION, groupSignup.getPrimarySubscriptionId()))
+                            .build()
+            )).getSubscriptionGroup();
+
+            // then
+            FullSubscriptionGroupResponse subscriptionGroup = SUBSCRIPTION_GROUPS_CONTROLLER.findSubscriptionGroup(groupSignup.getPrimarySubscriptionId().toString());
+            List<Integer> expectedIds = new ArrayList<>(groupSignup.getSubscriptionIds());
+            expectedIds.add(newSubscription.getId());
+            assertThat(subscriptionGroup.getSubscriptionIds())
+                    .hasSize(3)
+                    .containsExactlyInAnyOrderElementsOf(expectedIds);
+
+            assertThat(response.getCustomerId()).isEqualTo(groupCustomer.getId());
+            assertThat(response.getPaymentCollectionMethod()).isEqualTo(CollectionMethod.AUTOMATIC);
+            assertThat(response.getCreatedAt()).isNotNull()
+                    .isBefore(ZonedDateTime.now());
+            assertThat(response.getSubscriptionIds())
+                    .hasSize(3)
+                    .containsExactlyInAnyOrderElementsOf(expectedIds);
+
+            SubscriptionGroupPaymentProfile paymentProfile = response.getPaymentProfile();
+            assertThat(paymentProfile.getId()).isEqualTo(signupPP.getId());
+            assertThat(paymentProfile.getFirstName()).isEqualTo(signupPP.getFirstName());
+            assertThat(paymentProfile.getLastName()).isEqualTo(signupPP.getLastName());
+            assertThat(paymentProfile.getMaskedCardNumber()).isEqualTo(signupPP.getMaskedCardNumber());
+        }
+
+        @Test
+        void shouldReturn404IfSubscriptionNotFound() {
```

Comment:
ive managed to invoke other type of error
```
"HTTP Response Not OK. Status code: 422. Response: '{errors:{group:[{type:invalid,message:Target ID was invalid or missing for the group.}]}}'.
```
this is when `newSusbsciprionId` exists, but `subscriptionId` in GroupTarget does not. Maybe we could test it too?